contracts/okp4-cognitarium/src/error.rs

```diff
@@ -1,11 +1,70 @@
-use cosmwasm_std::StdError;
+use cosmwasm_std::{StdError, Uint128};
+use rio_turtle::TurtleError;
+use rio_xml::RdfXmlError;
 use thiserror::Error;
 
 #[derive(Error, Debug, PartialEq)]
 pub enum ContractError {
     #[error("{0}")]
     Std(#[from] StdError),
 
-    #[error("Not implemented.")]
-    NotImplemented,
+    #[error("{0}")]
+    ParseRDF(#[from] RDFParseError),
+
+    #[error("{0}")]
+    Store(#[from] StoreError),
+
+    #[error("Only the owner can perform this operation.")]
+    Unauthorized,
+}
+
+impl From<RdfXmlError> for ContractError {
+    fn from(value: RdfXmlError) -> Self {
+        RDFParseError::from(value).into()
+    }
+}
+
+impl From<TurtleError> for ContractError {
+    fn from(value: TurtleError) -> Self {
+        RDFParseError::from(value).into()
+    }
+}
+
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum StoreError {
+    #[error("Maximum triples number exceeded: {0}")]
+    TripleCount(Uint128),
+
+    #[error("Maximum byte size exceeded: {0}")]
+    ByteSize(Uint128),
+
+    #[error("Maximum triple byte size exceeded: {0} / {1}")]
+    TripleByteSize(Uint128, Uint128),
+
+    #[error("Maximum insert byte size exceeded: {0}")]
+    InsertDataByteSize(Uint128),
+
+    #[error("Maximum insert triple count exceeded: {0}")]
+    InsertDataTripleCount(Uint128),
```

Comment:
Why not put the maximum value allowed on each error like it's done on `TripleByreSize` ? 

```suggestion
    #[error("Maximum triples number exceeded: {0} / {1}")]
    TripleCount(Uint128, Uint128),

    #[error("Maximum byte size exceeded: {0} / {1}")]
    ByteSize(Uint128, Uint128),

    #[error("Maximum triple byte size exceeded: {0} / {1}")]
    TripleByteSize(Uint128, Uint128),

    #[error("Maximum insert byte size exceeded: {0} / {1}")]
    InsertDataByteSize(Uint128, Uint128),

    #[error("Maximum insert triple count exceeded: {0} / {1}")]
    InsertDataTripleCount(Uint128, Uint128),
```

Reply:
Actually the value displayed in the error is the limit, I found un-necessary to display the value exceeding the limit because the data being inserted with a stream, the stream is interrupted as soon as the limit is exceeded. The displayed exceeded value would always be `$limit + 1` which is not really useful,  and doesn't reflect the real value if the stream would have been fully processed.

Reply:
Yes, I understand, you're right, with a stream is not necessary.